src/lib/OS/SyscallHandler.cc

```diff
@@ -666,16 +735,65 @@ void SyscallHandler::handleSyscall() {
       stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {0ull}};
       break;
     }
+    case 233: {  // madvise
+      // madvise is not supported
+      // always return zero to indicate success
+      stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {0ull}};
+      break;
+    }
     case 235: {  // mbind
       // mbind is not supported due to all binaries being single threaded.
       // Always return zero to indicate success
       stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {0ull}};
       break;
     }
     case 261: {  // prlimit64
-      // TODO: Functionality temporarily omitted as it is unused within
-      // workloads regions of interest and not required for their simulation
-      stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {0ull}};
+      pid_t pid = currentInfo_.registerArguments[0].get<pid_t>();
+      int resource = currentInfo_.registerArguments[1].get<int>();
+      uint64_t newLimit = currentInfo_.registerArguments[2].get<uint64_t>();
+      uint64_t oldLimit = currentInfo_.registerArguments[3].get<uint64_t>();
+      int64_t retVal = 0;
+      if (pid != 0) {
+        // We only support changes to current process
+        retVal = -EPERM;
+      } else {
+        if (resource != RLIMIT_STACK) {
+          std::cout << "[SimEng:SyscallHandler] Un-supported resource used in "
+                       "prlimit64 syscall."
+                    << std::endl;
+          retVal = -EINVAL;
+        } else {
+          if (newLimit) {
+            // Update rlimit for Process
+            uint64_t physAddr =
+                OS_->handleVAddrTranslation(newLimit, currentInfo_.threadId);
+            if (masks::faults::hasFault(physAddr)) {
+              retVal = -EFAULT;
```

Comment:
Is this value consistent across macOS and Linux, If it is please resolve the comment. Thanks.

Reply:
Do you think these should be moved to Constants.hh for consistency?

Reply:
I don't think so. May cause confusion as the libary defined values will work